modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcMultiStatementSelfTest.java

```diff
@@ -422,11 +408,18 @@ public void testAutoCommitFalse() throws Exception {
     }
 
     @Test
-    @Disabled("https://issues.apache.org/jira/browse/IGNITE-21167")
-    public void testAutoCommitFalseWithEmptyTx() throws Exception {
-        String txErrMsg = "Transaction control statement cannot be executed within an external transaction";
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testAutoCommitFalseTxControlStatementsNotSupported() throws Exception {
+        String txErrMsg = "Transaction control statements are not supported when autocommit mode is disabled";
         conn.setAutoCommit(false);
-        assertThrowsSqlException(txErrMsg, () -> stmt.execute("START TRANSACTION; SELECT 1; COMMIT;"));
+        assertThrowsSqlException(txErrMsg, () -> stmt.execute("START TRANSACTION; SELECT 1; COMMIT"));
+        assertThrowsSqlException(txErrMsg, () -> stmt.execute("COMMIT"));
+        assertThrowsSqlException(txErrMsg, () -> stmt.execute("START TRANSACTION"));
+
+        boolean res = stmt.execute("SELECT 1;COMMIT");
```

Comment:
seems we already have such a tests, it need to be merged or compound somehow 
testAutoCommitFalseNonCompleted

Reply:
Yep,  I've just renamed it and refactored a bit